test/solidity-ibc/ICS20TransferTest.t.sol

```diff
@@ -773,13 +736,16 @@ contract ICS20TransferTest is Test {
         );
         assertEq(ack, ICS20Lib.SUCCESSFUL_ACKNOWLEDGEMENT_JSON);
 
-        // find and extract data from the ICS20ReceiveTransfer event
-        Vm.Log[] memory entries = vm.getRecordedLogs();
-        assertEq(entries.length, 4);
-        Vm.Log memory receiveTransferLog = entries[3];
-        assertEq(receiveTransferLog.topics[0], IICS20Transfer.ICS20ReceiveTransfer.selector);
+        (address erc20Address,) = ics20Transfer.getReceiveERC20AddressAndSource(
```

Comment:
If we need a way to get this information from the outside, there needs to be a different way than this since this function cannot be public

Reply:
Yes, we might actually want to emit this it seems. So maybe this event wasn't so unnecessary

Reply:
If we use events, this simplifies most of tests. The event should at least emit the contract address that is created when.  An alternative could be to read the state of _ibcDenomContracts. I do agree i would prefer to keep the `getReceiveERC20AddressAndSource` as private.